server/graphql/v2/mutation/OrderMutations.js

```diff
@@ -177,6 +186,35 @@ const orderMutations = {
         order = await order.update({ PaymentMethodId: newPaymentMethod.id });
       }
 
+      // amount
+      if (amount !== order.totalAmount) {
+        if (amount < 100) {
+          throw new Error('Invalid amount.');
+        }
+
+        // If using a tier, amount can never be less than the minimum amount, or else $1.00
+        if (amount < tier.minimumAmount) {
+          throw new Error('Amount is less than minimum value allowed for this Tier.');
+        }
+
+        order = await order.update({ totalAmount: amount });
+      }
+
+      // tier
+      // check if the tier is different from the previous tier
+      if (tier.legacyId !== order.TierId) {
+        // check if the tier is a named tier
+        if (tier.legacyId !== null) {
+          const tierInfo = await models.Tier.findByPk(tier.legacyId);
+
+          if (!tierInfo) {
+            throw new Error(`No tier found with tier id: ${tier.legacyId} for collective ${order.FromCollectiveId}`);
```

Comment:
Doesn't seems right to report `FromCollectiveId` it's `CollectiveId` that we are interested in I think.